sns.py

```diff
@@ -0,0 +1,58 @@
+def positive1(self, sqs_arn: str, topic_arn: str) -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    # create SNS topic, connect it to the Lambda, publish test message
+    sns_client.subscribe(TopicArn=topic_arn, Protocol='sqs',
+                         Endpoint=sqs_arn)
+
+def positive2(self, sqs_arn: str, topic_arn: str) -> None:
+    session = boto3.Session()
+    sns_client = session.client('sns')
+    response = sns_client.subscribe(
+        TopicArn=topic_arn,
+        Protocol='sqs',
+        Endpoint=sqs_arn,
+        ReturnSubscriptionArn=True)
+    return response
+
+def positive3() -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    return sns_client.create_topic(Name=TEST_TOPIC_NAME)
+
+def negative1() -> None:
+    session = boto3.Session()
+    sns_client = session.client('sns')
+    topic_info = sns_client.create_topic(Name=TEST_TOPIC_NAME)
+    sns_client.subscribe(TopicArn=topic_info['TopicArn'], Protocol='sqs',
+                         Endpoint=aws_stack.sqs_queue_arn(TEST_QUEUE_NAME_FOR_SNS))
+    test_value = short_uid()
+    sns_client.publish(TopicArn=topic_info['TopicArn'], Message='test message for SQS',
+                       MessageAttributes={'attr1': {'DataType': 'String', 'StringValue': test_value}})
+
+def negative2() -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    # create SNS topic, connect it to the Lambda, publish test message
+    num_events_sns = 3
+    response = sns_client.create_topic(Name='TEST_TOPIC_NAME')
+    sns_client.subscribe(TopicArn=response['TopicArn'], Protocol='lambda',
+                         Endpoint=aws_stack.lambda_function_arn('TEST_LAMBDA_NAME_STREAM'))
+    for i in range(0, num_events_sns):
+        sns_client.publish(TopicArn=response['TopicArn'], Message='test message %s' % i)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Because Amazon SNS is a distributed system, there might be some latency when it propagates information. Therefore, we recommend that you don&#39;t create resources and then immediately publish to them. To ensure that newly created topics receive messages, don&#39;t bind the `publish` operation with the `subscribe` or `create_topic` operations.

[Learn more](https://aws.amazon.com/sns/faqs/)